includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -8,1789 +8,154 @@
-							}
-							if ( isset( $_POST[ $post_key ] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
-								$_POST[ $post_key ] = $this->settings[ $key ];
-							}
-						}
-						unset( $json_settings[ $key ] );
-					}
-				} catch ( Exception $e ) {
-					$this->add_error( $e->getMessage() );
-				}
-			}
-		}
-
-		update_option( $this->get_option_key( true ), apply_filters( 'woocommerce_settings_api_sanitized_fields_' . $this->id, $this->settings ), 'yes' );
-	}
-
-	/**
-	 * Process a settings import from a json file.
-	 */
-	public function process_settings_import() {
-
-		if ( empty( $_POST['amazon_pay_action'] ) || 'import_settings' !== $_POST['amazon_pay_action'] ) {
-			return;
-		}
-		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-		if ( isset( $_POST['amazon_pay_import_nonce'] ) && ! wp_verify_nonce( $_POST['amazon_pay_import_nonce'], 'amazon_pay_import_nonce' ) ) {
-			return;
-		}
-		if ( ! current_user_can( 'manage_options' ) ) {
-			return;
-		}
-
-		if ( empty( $_FILES['import_file']['tmp_name'] ) ) {
-			wp_die( esc_html__( 'Please upload a file to import', 'woocommerce-gateway-amazon-payments-advanced' ) );
-		}
-		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-		$import_file = $_FILES['import_file'];
-
-		$this->process_settings_from_file( $import_file );
-		wp_safe_redirect( admin_url( 'admin.php?page=wc-settings&tab=checkout&section=' . $this->id ) );
-		exit;
-	}
+	protected $checkout_session;
 
 	/**
-	 * Generate Custom HTML.
+	 * Constructor
 	 */
-	public function generate_custom_html( $id, $conf ) {
-		$html = isset( $conf['html'] ) ? wp_kses_post( $conf['html'] ) : '';
-
-		if ( $html ) {
-			ob_start();
-			?>
-			</table>
-			<?php echo $html; ?>
-			<table class="form-table">
-			<?php
-			$html = ob_get_clean();
-		}
+	public function __construct() {
+		parent::__construct();
 
-		return $html;
+		// Init Handlers
+		add_action( 'wp_loaded', array( $this, 'init_handlers' ), 11 );
 	}
 
 	/**
-	 * Save private key.
+	 * Amazon Pay is available if the following conditions are met (on top of
+	 * WC_Payment_Gateway::is_available).
 	 *
-	 * @param string $private_key Private key PEM string.
+	 * 1) Gateway enabled
+	 * 2) Correctly setup
+	 * 2) In checkout pay page.
+	 *
+	 * @return bool
 	 */
-	private function save_private_key( $private_key ) {
-		$validate_private_key = openssl_pkey_get_private( $private_key );
-		if ( $validate_private_key ) {
-			update_option( WC_Amazon_Payments_Advanced_Merchant_Onboarding_Handler::KEYS_OPTION_PRIVATE_KEY, $private_key );
-			return true;
+	public function is_available() {
+		$is_available = ( 'yes' === $this->enabled );
+
+		if ( function_exists( 'is_checkout_pay_page' ) && is_checkout_pay_page() ) {
+			return $is_available;
```

Comment:
this return skip the filter

Reply:
Implementing this filter here may affect subscription functionality. Add a note to check this later. Good catch.